libtransmission/peer-mgr.cc

```diff
@@ -396,18 +453,25 @@ class tr_swarm
             /* noop */
             break;
 
-        case tr_peer_event::Type::ClientGotRej:
-            s->active_requests.remove(s->tor->piece_loc(event.pieceIndex, event.offset).block, peer);
-            break;
-
         case tr_peer_event::Type::ClientGotChoke:
-            s->active_requests.remove(peer);
+            s->active_requests.remove(msgs);
             break;
 
         case tr_peer_event::Type::ClientGotPort:
```

Comment:
Note to self: Add case if the peer sends a different listen port than the one we already know.

Reply:
103eba6